Userland/Utilities/mktemp.cpp

```diff
@@ -28,45 +28,40 @@ static char* generate_random_filename(const char* pattern)
         new_filename[i] = random_characters[(get_random<u32>() % (sizeof(random_characters) - 1))];
     }
 
-    return new_filename;
+    return StringView { new_filename };
 }
 
-static char* make_temp(const char* pattern, bool directory, bool dry_run)
+static ErrorOr<StringView> make_temp(String const pattern, bool directory, bool dry_run)
 {
     for (int i = 0; i < 100; ++i) {
-        char* path = generate_random_filename(pattern);
+        auto path = generate_random_filename(pattern);
         if (dry_run) {
-            struct stat stat;
-            auto rc = lstat(path, &stat);
-            if (rc < 0 && errno == ENOENT)
+            auto stat_or_error = Core::System::lstat(path);
+            if (stat_or_error.is_error() && stat_or_error.error().code() == ENOENT)
                 return path;
         } else if (directory) {
-            if (!mkdir(path, 0700))
-                return path;
+            TRY(Core::System::mkdir(path, 0700));
+            return path;
         } else {
-            auto fd = open(path, O_RDWR | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR);
-            if (fd >= 0) {
-                close(fd);
+            auto fd_or_error = Core::System::open(path, O_RDWR | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR);
+            if (!fd_or_error.is_error()) {
+                TRY(Core::System::close(fd_or_error.value()));
                 return path;
             }
         }
-        free(path);
```

Comment:
Removing the two calls to free is not correct, it causes the strdup'd strings to be leaked. Instead of wrapping those owning pointers in a StringView (which is not idiomatic), a better change would be to just use String and get rid of the calls to strdup.

Reply:
Still struggle to properly understand StringView. Changed to String as suggested, using StringBuilder to build the random filename.